Dockerfile

```diff
@@ -1,46 +1,64 @@
-FROM circleci/golang:1.12
-
-USER root
-WORKDIR /go/src/github.com/
-
-ENV PROTOBUF_VERSION="3.7.0"
-
-# Dowload given version of the protobuf binary
-RUN wget https://github.com/google/protobuf/releases/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip && \
-    unzip protoc-${PROTOBUF_VERSION}-linux-x86_64.zip -d /usr/local && \
-    rm protoc-${PROTOBUF_VERSION}-linux-x86_64.zip
-
-# Clone golang/protobuf repo reset to the latest tag and install
-RUN mkdir golang && cd golang && git clone https://github.com/golang/protobuf.git && \
-    cd protobuf && git reset --hard $(git rev-list --tags --max-count=1) && \
-    go install github.com/golang/protobuf/protoc-gen-go && \
-    cd ../../ && rm -rf golang
-
-# Clone gogo/protobuf repo reset to the latest tag and install
-RUN mkdir gogo && cd gogo && git clone https://github.com/gogo/protobuf.git && \
-    cd protobuf && git reset --hard $(git rev-list --tags --max-count=1) && \
-    go install github.com/gogo/protobuf/protoc-gen-gogofast && \
-    go install github.com/gogo/protobuf/protoc-gen-gogofaster && \
-    go install github.com/gogo/protobuf/protoc-gen-gogoslick && \
-    cd ../../ && rm -rf gogo
-
-# Install the latest version of the protoc-gen-govalidator plugin
-RUN go get github.com/mwitkow/go-proto-validators/protoc-gen-govalidators && \
-    rm -rf mwitkow
-
-# Install the latest version of the protoc-gen-uwpartner plugin
-RUN go get github.com/utilitywarehouse/protoc-gen-uwpartner && \
-    rm -rf utilitywarehouse
-
-# Clone golangci repo reset to the latest tag and install
-RUN mkdir golangci && cd golangci && git clone https://github.com/golangci/golangci-lint.git && \
-    cd golangci-lint && git reset --hard $(git rev-list --tags --max-count=1) && \
-    go install github.com/golangci/golangci-lint/cmd/golangci-lint && \
-    cd ../../ && rm -rf golangci
+FROM golang:1.12.4
+
+ENV PROTOBUF_VERSION="3.7.0" \
+    GOLANG_PROTOBUF_VERSION="1.3.1" \
+    GOGO_PROTOBUF_VERSION="1.2.1" \
+    VALIDATE_PROTOBUF_VERSION="0.0.14" \
+    VALIDATORS_PROTOBUF_VERSION="1f388280e944c97cc59c75d8c84a704097d1f1d6" \
+    UWPARTNER_PROTOBUF_VERSION="de4552500027969912fd801dcc5269a153b3fffe" \
+    GOLANGCI_LINT_VERSION="1.16.0" \
```

Comment:
My thinking with the linter and proto plugins in the current setup was that we will always get the latest release, so just triggering the rebuild would fetch the new versions. I get that this might be bit annoying with the linter, but for the other stuff we could maybe get around that by tagging the images. This approach is also fine, it's just a question about which behaviour do we want. This setup requires a bit more work as we need to manually maintain all of these versions...

Reply:
There have been problems with inconsistent tooling between Partner engineers. This makes it explicit what we support & local tooling can then match that.

Reply:
I do not feel strongly about this currently, so I can revert this change.

Reply:
fir enough 👍

Reply:
Another way to help deal with that is to enforce latest versions always (CI and local) - I suppose that may come with some caveats.. this is sort of like the go modules debate ;)